workspaces/optic-engine/src/interactions/result.rs

```diff
@@ -1,13 +1,14 @@
 use crate::events::http_interaction::HttpInteraction;
 use crate::learn_shape::TrailObservationsResult;
 use crate::shapes::{JsonTrail, ShapeDiffResult};
-use crate::state::endpoint::{PathComponentId, RequestId, ResponseId, ShapeId};
+use crate::state::endpoint::{PathComponentId, QueryParametersId, RequestId, ResponseId, ShapeId};
 use serde::{Deserialize, Serialize};
 use std::collections::hash_map::{DefaultHasher, HashMap};
 use std::hash::{Hash, Hasher};
 
 #[derive(Debug, Deserialize, Serialize, Hash)]
 pub enum InteractionDiffResult {
+  UnmatchedQueryParameters(UnmatchedQueryParameters),
```

Comment:
This new variant of `InteractionDiffResult`, together with the interaction trail embedded in it, will let the client discern this is a diff in the query parameters. That will become important once we add support for that in the UI diff flow, so useful to be aware of :)